Framework/Beamline/src/DetectorInfo.cpp

```diff
@@ -398,5 +378,56 @@ DetectorInfo::buildMergeIndices(const DetectorInfo &other) const {
   return merge;
 }
 
+// Indices returned here are the list of time indexes not to merge
+std::vector<bool>
+DetectorInfo::buildMergeSyncScanIndices(const DetectorInfo &other) const {
+  checkSizes(other);
+  std::vector<bool> merge(other.m_scanIntervals->size(), true);
+
+  for (size_t t1 = 0; t1 < other.m_scanIntervals->size(); ++t1) {
```

Comment:
Presumably this N*M loop is necessary? Sorting scan intervals doesn't help?

Reply:
This is the sync scan case, so this should rarely be larger than a few hundred each, and I suppose often one of them might be size one? @ianbush What do you think?

Reply:
Yes, as this for the sync scan this should not be too expensive, especially when comparing with the loop required for `buildMergeIndices`. The most common case would be to merge a big one with one of size one (making a scanning workspace), or a big one with a smaller one (merging D2B runs into a single workspace.